src/client.d.ts

```diff
@@ -1,194 +1,177 @@
-            topP?: number,
-            randomSeed?: number,
-            stream?: boolean,
-            /**
-             * @deprecated use safePrompt instead
-             */
-            safeMode?: boolean,
-            safePrompt?: boolean,
-            toolChoice?: ToolChoice,
-            responseFormat?: ResponseFormat
-        ): object;
-
-        listModels(): Promise<ListModelsResponse>;
-
-        chat(options: {
-            model: string;
-            messages: Array<{ role: string; name?: string, content: string | string[], tool_calls?: ToolCalls[]; }>;
-            tools?: Array<{ type: string; function:Function; }>; 
-            temperature?: number;
-            maxTokens?: number;
-            topP?: number;
-            randomSeed?: number;
-            /**
-             * @deprecated use safePrompt instead
-             */
-            safeMode?: boolean;
-            safePrompt?: boolean;
-            toolChoice?: ToolChoice;
-            responseFormat?: ResponseFormat; 
-        }): Promise<ChatCompletionResponse>;
-
-        chatStream(options: {
-            model: string;
-            messages: Array<{ role: string; name?: string, content: string | string[], tool_calls?: ToolCalls[]; }>;
-            tools?: Array<{ type: string; function:Function; }>;
-            temperature?: number;
-            maxTokens?: number;
-            topP?: number;
-            randomSeed?: number;
-            /**
-             * @deprecated use safePrompt instead
-             */
-            safeMode?: boolean;
-            safePrompt?: boolean;
-            toolChoice?: ToolChoice;
-            responseFormat?: ResponseFormat;
-        }): AsyncGenerator<ChatCompletionResponseChunk, void, unknown>;
-
-        embeddings(options: {
-            model: string;
-            input: string | string[];
-        }): Promise<EmbeddingResponse>;
-    }
-
-    export default MistralClient;
+export interface ModelPermission {
+  id: string;
+  object: 'model_permission';
+  created: number;
+  allow_create_engine: boolean;
+  allow_sampling: boolean;
+  allow_logprobs: boolean;
+  allow_search_indices: boolean;
+  allow_view: boolean;
+  allow_fine_tuning: boolean;
+  organization: string;
+  group: string | null;
+  is_blocking: boolean;
 }
+
+export interface Model {
+  id: string;
+  object: 'model';
+  created: number;
+  owned_by: string;
+  root: string | null;
+  parent: string | null;
+  permission: ModelPermission[];
+}
+
+export interface ListModelsResponse {
+  object: 'list';
+  data: Model[];
+}
+
+export interface Function {
+  name: string;
+  description: string;
+  parameters: object;
+}
+
+// TODO: this is not actually used, but would technically be a break change to remove
+export enum ToolType {
+  function = 'function',
+}
+
+export interface FunctionCall {
+  name: string;
+  arguments: string;
+}
+
+export interface ToolCalls {
+  id: string;
+  function: FunctionCall;
+}
+
+export enum ResponseFormats {
```

Comment:
I think we should also remove the enums from the types and change them to types. A simple example like this does not work because the enums are TS only:

```ts
import MistralClient, { ResponseFormats } from "@mistralai/mistralai";

const chatResponse = await mistralClient.chat({
    model: "mistral-large",
    temperature: 0.1,
    responseFormat: { type: ResponseFormats.json_object },
})`
```

This raises the error `'mistralai/src/client.js' does not provide an export named 'ResponseFormats'`